src/libServer/LookupServer.cpp

```diff
@@ -876,7 +881,58 @@ Json::Value LookupServer::GetBalance(const string& address) {
     throw JsonRpcException(RPC_MISC_ERROR, "Unable To Process");
   }
 }
+string LookupServer::GetEthCall(const Json::Value& _json) {
+  LOG_MARKER();
+  const auto& addr = JSONConversion::checkJsonGetEthCall(_json);
+  // Add mutex
+  uint64_t gasRemained =
+      2 * DS_MICROBLOCK_GAS_LIMIT;  // for now set total gas as twice the ds gas
+                                    // limit
+  bytes code{};
+  bool ret = false;
+  {
+    shared_lock<shared_timed_mutex> lock(
+        AccountStore::GetInstance().GetPrimaryMutex());
+    Account* contractAccount =
+        AccountStore::GetInstance().GetAccount(addr, true);
+    if (contractAccount == nullptr) {
+      throw JsonRpcException(RPC_INVALID_PARAMS, "Account does not exist");
+    }
+    code = contractAccount->GetCode();
+  }
+  Address fromAddr;
+  string result;
+  uint64_t amount{0};
+  try {
+    if (_json.isMember("fromAddr")) {
+      fromAddr = Address(_json["fromAddr"].asString());
+    }
+    if (_json.isMember("amount")) {
+      const auto& amount_str = _json["amount"].asString();
+      amount = strtoull(amount_str.c_str(), NULL, 0);
+    }
+    if (_json.isMember("gasLimit")) {
+      const auto& gasLimit_str = _json["gasLimit"].asString();
+      gasRemained = strtoull(gasLimit_str.c_str(), NULL, 0);
```

Comment:
Please see the comment above re gasLimit.  Don't let the user override the maximum by setting this to a huge number.

Reply:
Noted